components/idp.js

```diff
@@ -49,6 +48,25 @@ module.config(function($routeProvider, routeResolverProvider) {
       throw new Error('Not authenticated.');
     }
   }
+  /* @ngInject */
+  function redirectIfReferred(
+    $location, $window, $route, brSessionService, config) {
+    // return early if session is present
+    var wasReferred = ($location.search().referral === 'true');
+    var redirectAuto = ($location.search().auto === 'true');
+    return brSessionService.get()
+      .then(function(session) {
+        if(session.identity) {
+          var referred = wasReferred ? '?referral=true' : '';
+          if(redirectAuto) {
+            $window.location = document.referrer;
```

Comment:
$document has different semantics, just using document for a proper referral link.